flink-core/src/main/java/org/apache/flink/api/common/typeutils/CompatibilityResult.java

```diff
@@ -60,10 +62,10 @@
 	 *
 	 * @return a result that signals migration is necessary, also providing a convert deserializer.
 	 */
-	public static <T> CompatibilityResult<T> requiresMigration(TypeDeserializer<T> convertDeserializer) {
+	public static <T> CompatibilityResult<T> requiresMigration(@Nonnull TypeDeserializer<T> convertDeserializer) {
 		Preconditions.checkNotNull(convertDeserializer, "Convert deserializer cannot be null.");
 
-		return new CompatibilityResult<>(true, Preconditions.checkNotNull(convertDeserializer));
+		return new CompatibilityResult<>(true, convertDeserializer);
```

Comment:
Does `@Nonnull` always ensure that the argument is not null? We could simply keep the additional `Preconditions` check.